tests/production_pause_max_rev_blks_test.py

```diff
@@ -0,0 +1,166 @@
+    finalizerbNode = cluster.getNode(2)   # finalizer node for defproducerb
+
+    Print("Set finalizer policy and start transition to Savanna")
+    # Specifically, need to configure finalizer name for finalizerbNode as defproducerb
+    transId = cluster.setFinalizers(nodes=[node0, finalizerbNode], finalizerNames=["defproducera", "defproducerb"])
+    assert transId is not None, "setfinalizers failed"
+    assert cluster.biosNode.waitForTransFinalization(transId), f"setfinalizers transaction {transId} was not rolled into a LIB block"
+    assert cluster.biosNode.waitForLibToAdvance(), "LIB did not advance after setFinalizers"
+
+    # biosNode no longer needed
+    cluster.biosNode.kill(signal.SIGTERM)
+    cluster.waitOnClusterSync(blockAdvancing=5)
+
+    Print("Wait for LIB on all producing nodes to advance")
+    assert node0.waitForLibToAdvance(), "node0 did not advance LIB"
+    assert producerbNode.waitForLibToAdvance(), "producerbNode did not advance LIB"
+
+    ####################### test 1 ######################
+    # production paused due to max reversible blocks exceeded
+
+    Print("Shutdown finalizerbNode")
+    finalizerbNode.kill(signal.SIGTERM)
+    assert not finalizerbNode.verifyAlive(), "finalizerbNode did not shutdown"
+
+    # Verify LIB stalled on node0 and producerbNode due to finalizerNode was shutdown
+    Print("Verify LIB stalled  after shutdown of finalizerbNode")
+    if producerbNode.waitForLibToAdvance(timeout=5): # LIB can advance for a few blocks first
+        assert not producerbNode.waitForLibToAdvance(timeout=5), "LIB should not advance on producerbNode after finalizerbNode was shutdown"
+    if node0.waitForLibToAdvance(timeout=5): # LIB can advance for a few blocks first
+        assert not node0.waitForLibToAdvance(timeout=5), "LIB should not advance on node0 after finalizerbNode was shutdown"
+
+    # give time for max reversible blocks to be exceeded
+    time.sleep(24) # 24 is the time to produce 48 blocks which is the configured max-reversible-blocks
+
+    # Verify node0 and producerbNode then paused due to max reversible blocks exceeded
+    Print("Verify production paused after LIB stalled")
+    assert producerbNode.paused(), "producerbNode still producing after finalizerbNode was shutdown"
+    assert node0.paused(), "node0 still producing after finalizerbNode was shutdown"
+
+    ####################### test 2 ###########################
+    # nodeos can restart with a higher --max-reversible-blocks
+
+    node0.getInfo()
+    prevHeadBlockNum=node0.lastRetrievedHeadBlockNum
+
+    Print("Shutdown node0")
+    node0.kill(signal.SIGTERM)
+    assert not node0.verifyAlive(), "node0 did not shutdown"
+
+    addSwapFlags={"--max-reversible-blocks": "96"}
+    Print("Restart node0 with a higher max-reversible-blocks")
+    node0.relaunch(chainArg="--enable-stale-production", addSwapFlags=addSwapFlags)
+
+    Print("Wait for one round")
+    time.sleep(7)
```

Comment:
Instead of a sleep, wait for a head block above lib that exceeds max reversible.